addons/sourcemod/scripting/freak_fortress_2.sp

```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
+				//Were in a bracket.
+				if (bracket + 1 == sizeof(_operator))
+				{
+					decl String:bossName[32];
+					KvRewind(bKv);
+					KvGetString(bKv, "name", bossName, sizeof(bossName));
+					LogError("[FF2] %s has too many operators in operation! Returning malformDefault", bossName);
+					return malformDefault;
+				}
+				closeBracket = false;
+				bracket++;
+				_operator[bracket] = Operation_None;
+				sum[bracket] = 0.0;
+				break;
 			}
 			case ')':
 			{
-				valueReady=true;
-				mustClose=true;
-			}
-			case '\0':
-			{
-				valueReady=true;
-			}
-			case 'n', 'x':
-			{
-				usePlayers=true;
-			}
-			case '+':
-			{
-				_operator[parentheses]=1;
-				canAdd=true;
-			}
-			case '-':
-			{
-				_operator[parentheses]=2;
-				canAdd=true;
+				//Close bracket now!
+				closeBracket = true;
+				break;
 			}
-			case '*':
+			default:
 			{
-				_operator[parentheses]=3;
-				canAdd=true;
+				StrCat(value, sizeof(value), buffer);
+				break;
 			}
-			case '/':
+		}
+		if (i == length && !closeBracket && !foundOperator) //Handle constant operation(eg: "health_formula" 5000
+		{
+			foundOperator = true;
+		}
+		if (foundOperator || closeBracket)
+		{
+			if (foundOperator)
 			{
-				_operator[parentheses]=4;
-				canAdd=true;
+				foundOperator = false;
 			}
-			case '^':
+			if (closeBracket)
 			{
-				_operator[parentheses]=5;
-				canAdd=true;
+				closeBracket = false;
+				bracket--;
+				switch(_operator[bracket])
+				{
+					case Operation_Add:
+					{
+						sum[bracket] += sum[bracket+1];
+						break;
+					}
+					case Operation_Subtract:
+					{
+						sum[bracket] -= sum[bracket+1];
+						break;
+					}
+					case Operation_Multiplication:
+					{
+						sum[bracket] *= sum[bracket+1];
+						break;
+					}
+					case Operation_Division:
+					{
+						sum[bracket] /= sum[bracket+1];
+						break;
+					}
+					case Operation_Exponential:
+					{
+						sum[bracket] ^= sum[bracket+1];
+						break;
+					}
+					default:
+					{
+						sum[bracket] = sum[bracket+1];
+						break;
+					}
+				}
+				_operator[bracket+1] = Operation_None;
 			}
-			default:
+			if (usePlayers)
 			{
-				StrCat(value, sizeof(value), buffer);
+				usePlayers = false;
+				//Add sum with players.
+				switch(_operator[bracket])
+				{
+					case Operation_Add:
+					{
+						sum[bracket] += float(playing);
+						break;
+					}
+					case Operation_Subtract:
+					{
+						sum[bracket] -= float(playing);
+						break;
+					}
+					case Operation_Multiplication:
+					{
+						sum[bracket] *= float(playing);
+						break;
+					}
+					case Operation_Division:
+					{
+						sum[bracket] /= float(playing);
+						break;
+					}
+					case Operation_Exponential:
+					{
+						sum[bracket] ^= float(playing);
+						break;
+					}
+					default:
+					{
+						sum[bracket] = float(playing);
+						break;
+					}
+				}
 			}
-		}
-
-		if(valueReady)
-		{
-			valueReady=false;
-			if(usePlayers)
+			if (useLives)
```

Comment:
I always felt it be easier for boss makers if ff2 handles doing math involving lives